sormas-ui/src/main/java/de/symeda/sormas/ui/events/EventsFilterForm.java

```diff
@@ -130,6 +142,64 @@ public void addMoreFilters(CustomLayout moreFiltersContainer) {
 
 		moreFiltersContainer.addComponent(buildWeekAndDateFilter(EventCriteria.DateType.EVENT), EVENT_WEEK_AND_DATE_FILTER);
 		moreFiltersContainer.addComponent(buildWeekAndDateFilter(EventCriteria.DateType.ACTION), ACTION_WEEK_AND_DATE_FILTER);
+
+		ComboBox facilityTypeGroup = new ComboBox();
+		facilityTypeGroup.setId(FACILITY_TYPE_GROUP_FILTER);
+		facilityTypeGroup.setInputPrompt(I18nProperties.getCaption(Captions.Facility_typeGroup));
+		facilityTypeGroup.addStyleNames(ValoTheme.OPTIONGROUP_HORIZONTAL, CssStyles.OPTIONGROUP_CAPTION_INLINE, CssStyles.OPTIONGROUP_GRID_LAYOUT);
+		facilityTypeGroup.setWidth(140, Unit.PIXELS);
+		facilityTypeGroup.addItems((Object[]) FacilityTypeGroup.values());
+		facilityTypeGroup.setVisible(false);
+		moreFiltersContainer.addComponent(facilityTypeGroup, FACILITY_TYPE_GROUP_FILTER);
+
+		ComboBox facilityType = addField(
+			moreFiltersContainer,
+			FieldConfiguration.withCaptionAndPixelSized(
+				LocationDto.FACILITY_TYPE,
+				I18nProperties.getPrefixCaption(LocationDto.I18N_PREFIX, LocationDto.FACILITY_TYPE),
+				140));
+		facilityType.setVisible(false);
+
+		ComboBox facility = addField(
+			moreFiltersContainer,
+			FieldConfiguration
+				.withCaptionAndPixelSized(LocationDto.FACILITY, I18nProperties.getPrefixCaption(LocationDto.I18N_PREFIX, LocationDto.FACILITY), 140));
+		facility.setEnabled(false);
+		facility.setVisible(false);
+
+		getField(EventDto.TYPE_OF_PLACE).addValueChangeListener(e -> {
+			boolean visible = e.getProperty().getValue() == TypeOfPlace.FACILITY;
+			if (!visible) {
+				facility.clear();
+				facilityType.clear();
+				facilityTypeGroup.clear();
+			}
+			facility.setVisible(visible);
+			facilityType.setVisible(visible);
+			facilityTypeGroup.setVisible(visible);
+		});
+
+		facilityTypeGroup.addValueChangeListener(
+			e -> FieldHelper.updateEnumData(
+				facilityType,
+				facilityTypeGroup.getValue() != null
+					? FacilityType.getTypes((FacilityTypeGroup) facilityTypeGroup.getValue())
+					: Arrays.stream(FacilityType.values()).collect(Collectors.toList())));
+
+		facilityType.addValueChangeListener(e -> {
+			if (facilityType.getValue() != null) {
+				if (facilityTypeGroup.getValue() == null) {
+					facilityTypeGroup.setValue(((FacilityType) facilityType.getValue()).getFacilityTypeGroup());
+				}
+				facility.setEnabled(true);
+				FieldHelper.updateItems(
+					facility,
+					FacadeProvider.getFacilityFacade().getActiveFacilitiesByType((FacilityType) facilityType.getValue(), true, false));
```

Comment:
If I'm not mistaken this will pull all facilities that are in the system out of the database irrespective of the selected region, district and community; that's not a good idea because it could be a huge amount; the region, district and community filters should be respected (and required) for this field and its values

Reply:
I added a dependency to district/community.